genienlp/models/base.py

```diff
@@ -409,6 +409,11 @@ def validate_e2e_dialogues(
         device = self.device
         args = self.args
 
+        if self.numericalizer._tokenizer.src_lang:
+            src_lang = self.numericalizer._tokenizer.src_lang
+        else:
+            src_lang = self.orig_src_lang
```

Comment:
It seems to me that `self.orig_src_lang` can be `None` in this line for HF tokenizers that don't have language codes. Is `src_lang` part of their API and therefore guaranteed for all tokenizers? If not, wouldn't it break `lang=src_lang[:2]` in line 534, or similarly in line 194?

Reply:
we set `src_lang` and `tgt_lang` attribute for all the tokenizers here: https://github.com/stanford-oval/genienlp/blob/fb35befc2ed2a1c606b9b1dbbb7c3c5df1a46b7e/genienlp/data_utils/numericalizer.py#L216
`src_lang` is the adjusted version of `orig_src_lang`: https://github.com/stanford-oval/genienlp/blob/fb35befc2ed2a1c606b9b1dbbb7c3c5df1a46b7e/genienlp/util.py#L740
`orig_src_lang` is only used for Marian tokenizers. For single-language Marian models (those that source and target language are only one language) we set `src_lang` and `tgt_lang` to None since they don't need it. Check this code for more info: https://github.com/stanford-oval/genienlp/blob/fb35befc2ed2a1c606b9b1dbbb7c3c5df1a46b7e/genienlp/models/transformer_seq2seq.py#L60